allure-framework-modules/allure-security-module/src/main/java/com/capgemini/ntc/security/core/BasePage.java

```diff
@@ -0,0 +1,101 @@
+package com.capgemini.ntc.security.core;
+
+import com.capgemini.ntc.security.core.base.properties.PropertiesFileSettings;
+import com.capgemini.ntc.security.core.base.runtime.RuntimeParameters;
+import com.capgemini.ntc.test.core.BaseTest;
+import com.capgemini.ntc.test.core.BaseTestWatcher;
+import com.capgemini.ntc.test.core.ITestObserver;
+import com.capgemini.ntc.test.core.ModuleType;
+import com.capgemini.ntc.test.core.analytics.IAnalytics;
+import com.capgemini.ntc.test.core.base.environment.EnvironmentModule;
+import com.capgemini.ntc.test.core.base.environment.IEnvironmentService;
+import com.capgemini.ntc.test.core.base.properties.PropertiesSettingsModule;
+import com.capgemini.ntc.test.core.logger.BFLogger;
+import com.google.inject.Guice;
+
+abstract public class BasePage implements ITestObserver {
+	
+	private final static PropertiesFileSettings	propertiesFileSettings;
+	private static IEnvironmentService					environmentService;
+	private final static IAnalytics							analytics;
+	public final static String									analitycsCategoryName	= "Security-Module";
+	
+	static {
+		// Get analytics instance created in BaseTets
+		analytics = BaseTest.getAnalytics();
+		
+		// Get and then set properties information from selenium.settings file
+		propertiesFileSettings = createPropertiesSettings();
+		
+		// Read System or maven parameters
+		setRuntimeParametersSelenium();
+		
+		// Read Environment variables either from environmnets.csv or any other input data.
+		setEnvironmetInstance();
+	}
+	
+	public static IAnalytics getAnalytics() {
+		return BasePage.analytics;
+	}
+	
+	public BasePage() {
+		this.addObserver();
+	}
+	
+	@Override
+	public void addObserver() {
+		BaseTestWatcher.addObserver(this);
+	}
+	
+	@Override
+	public void onTestFailure() {
+		BFLogger.logDebug("BasePage.onTestFailure    " + this.getClass()
+		        .getSimpleName());
+	}
+	
+	@Override
+	public void onTestSuccess() {
+		// All actions needed while test method is success
+		BFLogger.logDebug("BasePage.onTestSuccess    " + this.getClass()
+		        .getSimpleName());
+	}
+	
+	@Override
+	public void onTestFinish() {
+		// All actions needed while test class is finishing
+		BFLogger.logDebug("BasePage.onTestFinish   " + this.getClass()
+		        .getSimpleName());
+		BaseTestWatcher.removeObserver(this);
+	}
+	
+	@Override
+	public void onTestClassFinish() {
+		BFLogger.logDebug("BasePage.onTestClassFinish   " + this.getClass()
+		        .getSimpleName());
+	}
+	
+	@Override
+	public ModuleType getModuleType() {
+		return ModuleType.EXAMPLE;
```

Comment:
Please switch to ModuleType.SECURITY

Reply:
This enumeration value is not present right now. Shall I extend it?

Reply:
Yes. Please extend it, and then update allure-core-module version to 4.12.0.5-SNAPSHOT